src/RahaStellar.ts

```diff
@@ -0,0 +1,128 @@
+import * as bs58 from 'bs58';
+import { Buffer } from 'buffer';
+import fetch from 'isomorphic-fetch';
+import StellarSdk from 'stellar-sdk';
+import { URL, URLSearchParams } from 'url';
+
+import { IPFS_SHA_256_LEN_32_PREFIX, STELLAR_ENDPOINT, RAHA_IO_STELLAR_PUBLIC_KEY, STELLAR_TEST_ENDPOINT } from './constants';
+
+/**
+ * Note: TransactionMetadata is undocumented and quite ugly to interact with. Mark had to look into the SDK source to do
+ * TODO: Can we use documented features/do this in an easier to understand and less brittle way?
+ */
+function getFirstOperationAttributesFromTransactionMeta(transactionMeta) {
+    return transactionMeta._value[0]._attributes.changes[0]._value._attributes.data._value._attributes
+}
+
+/**
+ * Return the name of a block from transaction metadata.
+ */
+export function getBlockNameFromTransactionMeta(transactionMeta) {
+    return getFirstOperationAttributesFromTransactionMeta(transactionMeta).dataName;
+}
+
+/**
+ * Legacy method for blocks set using only managed data with no memo hash.
+ * See note on TransactionMetadata above.
+ */
+export function getMultiHashFromTransactionMeta(transactionMeta) {
+    const charCode = getFirstOperationAttributesFromTransactionMeta(transactionMeta).dataValue;
+    return String.fromCharCode.apply(null, charCode);
+}
+
+/**
+ * Extract the TransactionMeta XDR object from the Stellar transaction.
+ * See note on TransactionMetadata above.
+ */
+export function getTransactionMetaFromTransaction(stellarTx) {
+    // xdr cast because StellarSdk XDR typing is bad.
+    return (StellarSdk.xdr as any).TransactionMeta.fromXDR(stellarTx.result_meta_xdr, 'base64');
+}
+
+export function sha256MultiHashToMemo(base58MultiHash: string) {
+    const multiHashBuffer = bs58.decode(base58MultiHash);
+    const multiHashBytes = Uint8Array.from(multiHashBuffer);
+    for (let i = 0; i < IPFS_SHA_256_LEN_32_PREFIX.length; ++i) {
+        if (i >= multiHashBytes.length || IPFS_SHA_256_LEN_32_PREFIX[i] !== multiHashBytes[i]) {
+            throw new Error('Invalid MultiHash prefix.');
+        }
+    }
+    return multiHashBuffer.slice(2);
+}
+
+/**
+ * Create a new Stellar account for testing (on the test network). Returns a
+ * KeyPair object with the new account's public and private keys.
+ */
+export async function getNewTestAccount() {
+    const keyPair = StellarSdk.Keypair.random();
+    const friendBotFundRequest = new URL('https://friendbot.stellar.org');
+    friendBotFundRequest.searchParams.append('addr', keyPair.publicKey());
+    try {
+        await fetch(friendBotFundRequest.href);
+        return keyPair;
+    } catch (err) {
```

Comment:
Catching then throwing seems like unnecessary code complexity, replace 61-67 with `return await fetch(friendBotFundRequest.href);` ?

Reply:
Agreed, I think catching and throwing makes sense if you’re trying to wrap the underlying error with a client-understandable one, but most of the time it’s not necessary; if you want to console log something then do that log in the caller

Reply:
kk. removed.